.github/workflows/pr-ci.yml

```diff
@@ -73,9 +73,42 @@ jobs:
     with:
       checkout-repo: ${{ needs.checkout-branch.outputs.PR_repo }}
       checkout-ref: ${{ needs.checkout-branch.outputs.PR_head_ref }}
+
+  start-comment:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Capture Date and Time
+        id: date-time
+        run: echo "::set-output name=date-time::$(date +'%Y-%m-%d %H:%M:%S')"
+      - name: Start Comment
+        uses: thollander/actions-comment-pull-request@v2
+        with:
+          message: |
+            Workflow started at ${{steps.date-time.outputs.date-time}}.
       
   build-and-test:
     needs: [code-analysis, checkout-branch]
+    strategy:
+      matrix:
+        arch: [amd64, arm64]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: build-test
+        uses: ./.github/workflows/ci-build-image.yml
+        with:
+         build-arch: ${{ matrix.arch }}
+         checkout-repo: ${{ needs.checkout-branch.outputs.PR_repo }}
+         checkout-ref: ${{ needs.checkout-branch.outputs.PR_head_ref  }}
+         skip-itests: ${{ matrix.arch != 'amd64' }}
+      - name: Set BUILD_TEST_EXECUTED flag
+        id: set-build-test-executed
+        run: echo "BUILD_TEST_EXECUTED=true" >> $GITHUB_ENV
+
+  retest-integration:
```

Comment:
I'm pretty sure you don't intend on rebuilding the image here? Might need to add an input to ci-build-image.yml to specify that you only want to run integration tests.

Reply:
Or perhaps it would be better to create a separate reusable workflow that pulls an already-built image tag from ghcr.io and runs the integration test suite against it?